engine/src/main/java/io/camunda/zeebe/engine/Engine.java

```diff
@@ -85,37 +92,38 @@ public void replay(final TypedRecord event) {
   @Override
   public ProcessingResult process(
       final TypedRecord record, final ProcessingResultBuilder processingResultBuilder) {
-    TypedRecordProcessor<?> currentProcessor = null;
-
-    final var typedCommand = (TypedRecord<?>) record;
-    try {
-      currentProcessor =
-          recordProcessorMap.get(
-              typedCommand.getRecordType(),
-              typedCommand.getValueType(),
-              typedCommand.getIntent().value());
-    } catch (final Exception e) {
-      LOG.error(ERROR_MESSAGE_ON_EVENT_FAILED_SKIP_EVENT, typedCommand, e);
-    }
+    try (final var scope = new ProcessingResultBuilderScope(processingResultBuilder)) {
```

Comment:
❓ Why do we need this? I see it sets the mutex, but I don't understand how this mutex is working here. Does this try with resources somehow only lets 1 thread access the resource at a time?

Reply:
We don't need this. And it doesn't address multthreading. The only thing the try-with-resources does is to close the scope even if an exception is thrown in the block